src/main/java/core/basesyntax/Figure.java

```diff
@@ -0,0 +1,28 @@
+package core.basesyntax;
+
+public abstract class Figure {
+    private String color;
+
+    public Figure(String color) {
+        this.color = color;
+    }
+
+    public abstract double getArea();
+
+    public String getColor() {
+        return color;
+    }
+
+    public void setColor(String color) {
+        this.color = color;
+    }
+
+    public void draw() {
+        moreInfo();
+        System.out.println("Area: " + getArea() + " sq. units");
+        System.out.println("Color: " + color);
+        System.out.println();
+    }
+
+    protected abstract void moreInfo();

```

Comment:
The `moreInfo()` method is abstract here, which is fine, but make sure that this method is conceptually similar to what the class represents. If it's not, it might be better to have it in a different interface or class structure.